apps/condo/domains/user/components/auth/ValidatePhoneForm.tsx

```diff
@@ -143,13 +143,14 @@ export const ValidatePhoneForm = ({ onFinish, onReset, title }): React.ReactElem
         }
     }, [confirmPhone])
 
-    useEffect(() => {
-        if (isPhoneVisible) {
-            setShowPhone(formatPhone(phone))
-        } else {
-            const unHidden = formatPhone(phone)
-            setShowPhone(`${unHidden.substring(0, 9)}***${unHidden.substring(12)}`)
-        }
+    const handleNumberVisible = useCallback(() => {
+        const formattedPhone = formatPhone(phone)
+        isPhoneVisible
```

Comment:
please use ternary operator only for providing values, in this case you need to use classical `if`

Reply:
thou you're calling the same functions, other possibility here is to precalculate value using ternary operator and then call the function once.

Reply:
```
const phoneValue = isPhoneVisible ? formattedPhone : `${formattedPhone.substring(0, 9)}***${formattedPhone.substring(12)}`

setShowPhone(phoneValue)
```